plugins/org.python.pydev/tests/org/python/pydev/ast/interpreter_managers/InterpreterInfoTest.java

```diff
@@ -323,4 +330,86 @@ public void testInterpreterInfoOutputWithGarbageBeforeAfterXML() throws Exceptio
     }
 
+    public void testObtainCondaEnv() throws Exception {
+        CorePlugin.setBundleInfo(new BundleInfoStub());
+
+        String testFileName = "env_vars.sh";
+        if (TestDependent.isWindows()) {
+            testFileName = "env_vars.bat";
+        }
+
+        String activateFilePath = TestDependent.CONDA_PYTHON_ENV + "etc/conda/activate.d/" + testFileName;
+        String deactivateFilePath = TestDependent.CONDA_PYTHON_ENV + "etc/conda/deactivate.d/" + testFileName;
+
+        String testVariable = "MY_INTERPRETERINFO_TEST_KEY";
+
+        Tuple<String, String> contents = getCondaActivationAndDeactivationTestContent(testVariable);
+
+        writeFile(activateFilePath, contents.o1);
+        writeFile(deactivateFilePath, contents.o2);
+
+        String pythonExe = TestDependent.CONDA_PYTHON_ENV + "bin/python";
+        if (TestDependent.isWindows()) {
+            pythonExe = TestDependent.CONDA_PYTHON_ENV + "python.exe";
+        }
+        InterpreterInfo interpreterInfo = new InterpreterInfo("3.8", pythonExe, new ArrayList<String>());
+        File condaPrefix = interpreterInfo.getCondaPrefix();
+        Map<String, String> condaEnv = interpreterInfo.obtainCondaEnv(condaPrefix);
+
+        if (!condaEnv.containsKey(testVariable)) {
+            throw new AssertionFailedError("Expected variable `" + testVariable + "` to exist in conda `"
+                    + condaPrefix.getAbsolutePath() + "` environment");
+        }
+    }
+
+    private Tuple<String, String> getCondaActivationAndDeactivationTestContent(String key) {
+        if (TestDependent.isWindows()) {
+            String activationContent = "set " + key + "='interpreterinfo-test-value'";
+            String deactivationContent = "set " + key + "=";
+            return new Tuple<String, String>(activationContent, deactivationContent);
+        } else {
+            String header = "#!/bin/sh\n\n";
+            String activationContent = header + "export " + key + "='interpreterinfo-test-value'";
+            String deactivationContent = header + "unset " + key;
+            return new Tuple<String, String>(activationContent, deactivationContent);
+        }
+    }
+
+    private File writeFile(String path, String content) throws Exception {
+        File file = new File(path);
+        if (!file.exists()) {
+            makeDirs(file.getParent());
+            boolean isCreated = file.createNewFile();
+            if (!isCreated) {
+                throw new Exception("Could not create file " + file.getAbsolutePath());
+            }
+        }
+        writeContentInFile(content, file);
+        return file;
+    }
+
+    private void makeDirs(String path) throws Exception {
+        File dir = new File(path);
+        boolean isDirsCreated = dir.mkdirs();
+        if (!isDirsCreated) {
+            throw new Exception("Could create dirs from " + path + " path");
+        }
+    }
+
+    private void writeContentInFile(String content, File file) {
```

Comment:
Should use: `org.python.pydev.shared_core.io.FileUtils.writeStrToFile(String, File)` instead.